src/Cli/Utils.cs

```diff
@@ -608,13 +610,20 @@ public static bool TryGetRoleAndOperationFromPermission(IEnumerable<string> perm
             [NotNullWhen(true)] out RuntimeConfig? deserializedRuntimeConfig)
         {
             deserializedRuntimeConfig = null;
-            if (!TryReadRuntimeConfig(configFile, out string runtimeConfigJson))
+            string? runtimeConfigJson;
+
+            try
+            {
+                // Tries to read the config and resolve environment variables.
+                runtimeConfigJson = GetRuntimeConfigJsonString(configFile);
+            }
+            catch (Exception e)
             {
-                _logger.LogError($"Failed to read the config file: {configFile}.");
+                _logger.LogError("Failed due to: {exceptionMessage}", e.Message);
```

Comment:
Might be worth mentioning "What failed?"

Reply:
it woukld be printing the same thing twice, the method called in the condition already logs the exception